From c7db95e74d9ff6d563d1627f283f3d435a595046 Mon Sep 17 00:00:00 2001
From: Jason Montleon <jmontleo@redhat.com>
Date: Tue, 10 Jun 2025 14:33:43 -0400
Subject: [PATCH 1234/1239] Revert "insmod: simplify section header process for
 optimize insmod speed"

This reverts commit c5737d27d65528e4b183a45cc684b34d5cbde981.

If I understand correctly this is trying to achieve the same effect as
the conflicting 6.16 commit be17c0df67959fe4f88dac75dc26ed9252d4b133
---
 arch/riscv/kernel/module-sections.c | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/arch/riscv/kernel/module-sections.c b/arch/riscv/kernel/module-sections.c
index 5964d22aba644..1675cbad8619d 100644
--- a/arch/riscv/kernel/module-sections.c
+++ b/arch/riscv/kernel/module-sections.c
@@ -55,8 +55,6 @@ unsigned long module_emit_plt_entry(struct module *mod, unsigned long val)
 
 	return (unsigned long)&plt[i];
 }
-/* it seems that the duplicate check is not necessary */
-#ifndef CONFIG_SOC_SPACEMIT
 
 #define cmp_3way(a, b)	((a) < (b) ? -1 : (a) > (b))
 
@@ -71,7 +69,6 @@ static int cmp_rela(const void *a, const void *b)
 		i = cmp_3way(x->r_addend, y->r_addend);
 	return i;
 }
-#endif
 
 static bool duplicate_rela(const Elf_Rela *rela, int idx)
 {
@@ -92,17 +89,9 @@ static void count_max_entries(const Elf_Rela *relas, size_t num,
 		switch (ELF_R_TYPE(relas[i].r_info)) {
 		case R_RISCV_CALL_PLT:
 		case R_RISCV_PLT32:
-#ifndef CONFIG_SOC_SPACEMIT
-/* it seems that the duplicate check is not necessary */
-			if (!duplicate_rela(relas, i))
-#endif
 			(*plts)++;
 			break;
 		case R_RISCV_GOT_HI20:
-#ifndef CONFIG_SOC_SPACEMIT
-/* it seems that the duplicate check is not necessary */
-			if (!duplicate_rela(relas, i))
-#endif
 			(*gots)++;
 			break;
 		default:
-- 
2.53.0

